src/main/java/net/minecraftforge/client/gui/ScrollPanel.java

```diff
@@ -63,12 +72,41 @@ public ScrollPanel(Minecraft client, int width, int height, int top, int left)
         this.bottom = height + this.top;
         this.right = width + this.left;
         this.barLeft = this.left + this.width - barWidth;
+        this.border = border;
+        this.barWidth = barWidth;
+        this.bgColorFrom = bgColorFrom;
+        this.bgColorTo = bgColorTo;
+        this.barBgColor = barBgColor;
+        this.barColor = barColor;
+        this.barBorderColor = barBorderColor;
     }
 
     protected abstract int getContentHeight();
 
     protected void drawBackground() {}
 
+    protected void drawBackground(PoseStack matrix, Tesselator tess, float partialTicks)
+    {
+        BufferBuilder worldr = tess.getBuilder();
+
+        if (this.client.level != null)
+        {
+            this.drawGradientRect(matrix, this.left, this.top, this.right, this.bottom, bgColorFrom, bgColorTo);
+        }
+        else // Draw dark dirt background
+        {
+            RenderSystem.setShader(GameRenderer::getPositionTexColorShader);
+            RenderSystem.setShaderTexture(0, GuiComponent.BACKGROUND_LOCATION);
+            final float texScale = 32.0F;
+            worldr.begin(VertexFormat.Mode.QUADS, DefaultVertexFormat.POSITION_TEX_COLOR);
+            worldr.vertex(this.left,  this.bottom, 0.0D).uv(this.left  / texScale, (this.bottom + (int)this.scrollDistance) / texScale).color(0x20, 0x20, 0x20, 0xFF).endVertex();
+            worldr.vertex(this.right, this.bottom, 0.0D).uv(this.right / texScale, (this.bottom + (int)this.scrollDistance) / texScale).color(0x20, 0x20, 0x20, 0xFF).endVertex();
+            worldr.vertex(this.right, this.top,    0.0D).uv(this.right / texScale, (this.top    + (int)this.scrollDistance) / texScale).color(0x20, 0x20, 0x20, 0xFF).endVertex();
+            worldr.vertex(this.left,  this.top,    0.0D).uv(this.left  / texScale, (this.top    + (int)this.scrollDistance) / texScale).color(0x20, 0x20, 0x20, 0xFF).endVertex();
+            tess.end();
+        }
+    }
+
     /**
      * Draw anything special on the screen. GL_SCISSOR is enabled for anything that
      * is rendered outside of the view box. Do not mess with SCISSOR unless you support this.
```

Comment:
Outdated javadoc.

Reply:
no because I just replaced the GL11 things with the RenderSystem wrappers (it still does the same thing)

Reply:
It still is outdated. GL_SCISSOR is internally called, but we don't want people to reference GL calls directly so it should mention the method at least instead.